src/exo/LoopIR_scheduling.py

```diff
@@ -2564,11 +2564,147 @@ def map_e(self, e):
         return e
 
 
+
+class _DoNormalize(LoopIR_Rewrite):
+    def __init__(self, proc):
+        self.C = Sym("temporary_constant_symbol")
+        super().__init__(proc)
+
+        self.proc = InferEffects(self.proc).result()
+
+    def concat_map(self, op, lhs, rhs):
```

Comment:
It would be good to have a comment here or somewhere about what these `dict`s that we're operating on actually mean.

i.e. the dictionaries actually represent linear combinations of symbols, which are the keys.  The coefficient for a particular symbol is the value that key maps to.  One option for factoring this code is that you could choose to create an "affine expression" class that maintains the dictionary and implements the relevant operations.

Reply:
If you do have a class, the class could also manage the conversion of itself back to `LoopIR.expr`s

Reply:
I'll add comments, thanks!